src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -131,6 +139,46 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
             executionContext.Debug("Diagnostic file upload complete.");
         }
 
+        private async Task DumpCloudInitLogs(DateTime jobStartTimeUtc, string diagFolder)
+        {
+            var builder = new StringBuilder();
+
+            string cloudInit = WhichUtil.Which("cloud-init", trace: Trace);
```

Comment:
Do we need to check presence of cloud-init in Path environment variable (this will be done by WhichUtil.Which method)?

Reply:
As far as I see `WhichUtil.Which` is the safe way to check availability of cloud-init and to find out the full path to it. Or not recommend using this tool?